fractal_server/app/routes/api/v2/project_share.py

```diff
@@ -0,0 +1,305 @@
+        )
+        .join(LinkUserProjectV2, LinkUserProjectV2.user_id == UserOAuth.id)
+        .where(LinkUserProjectV2.project_id == project_id)
+        .where(LinkUserProjectV2.is_owner.is_(False))
+    )
+    links = res.scalars().all()
+
+    return [
+        ProjectShareRead(
+            user_email=link[0],
+            is_verified=link[1],
+            permissions=link[2],
+        )
+        for link in links
+    ]
+
+
+@router.post("/project/{project_id}/link/", status_code=201)
+async def share_a_project(
+    project_id: int,
+    email: EmailStr,
+    project_invitation: ProjectShareCreate,
+    user: UserOAuth = Depends(current_user_act_ver_prof),
+    db: AsyncSession = Depends(get_async_db),
+):
+    # Check current user is project owner
+    await check_user_is_project_owner(
+        user_id=user.id, project_id=project_id, db=db
+    )
+
+    # Get the ID of the user to invite
+    invited_user_id = await get_user_id_from_email(user_email=email, db=db)
+
+    # Check if link already exists
+    await check_user_is_not_linked_to_project(
+        user_id=invited_user_id,
+        project_id=project_id,
+        db=db,
+    )
+
+    # Create new link
+    db.add(
+        LinkUserProjectV2(
+            project_id=project_id,
+            user_id=invited_user_id,
+            is_owner=False,
+            is_verified=False,
+            permissions=project_invitation.permissions,
+        )
+    )
+    await db.commit()
+
+    return
+
+
+@router.patch("/project/{project_id}/link/", status_code=200)
+async def patch_project_permissions(
+    project_id: int,
+    email: EmailStr,
+    update: ProjectShareUpdate,
+    user: UserOAuth = Depends(current_user_act_ver_prof),
+    db: AsyncSession = Depends(get_async_db),
+):
+    # Check current user is project owner
+    await check_user_is_project_owner(
+        user_id=user.id, project_id=project_id, db=db
+    )
+
+    # Get the ID of the linked user
+    linked_user_id = await get_user_id_from_email(user_email=email, db=db)
+
+    # Check you're not changing your own permissions
+    check_not_owner_id(user_id=linked_user_id, owner_id=user.id)
+
+    # Get the link to patch
+    link = await check_user_is_linked_to_project(
+        user_id=linked_user_id,
+        project_id=project_id,
+        db=db,
+    )
+
+    # Update and commit
+    for key, value in update.model_dump(exclude_unset=True).items():
+        setattr(link, key, value)
+    await db.commit()
+
+    return
+
+
+@router.delete("/project/{project_id}/link/", status_code=204)
+async def kick_out_guest(
+    project_id: int,
+    email: EmailStr,
+    user: UserOAuth = Depends(current_user_act_ver_prof),
+    db: AsyncSession = Depends(get_async_db),
+):
+    # Check current user is project owner
+    await check_user_is_project_owner(
+        user_id=user.id, project_id=project_id, db=db
+    )
+
+    # Get the ID of the linked user
+    linked_user_id = await get_user_id_from_email(user_email=email, db=db)
```

Comment:
What about using the "guest" keyword (here and in other endpoints)? (e.g. this one could be `guest_user_id`)